app/models/page.rb

```diff
@@ -39,5 +40,13 @@ def plugins
       plugin_class.where(page_id: id).to_a
     end.flatten
   end
+
+  private
+
+  def switch_layouts
+    if changed.include?("liquid_layout_id") || changed.include?("liquid_layout")
```

Comment:
Set intersection is a little less verbose (but is it necessary to check against `liquid_layout` as well?
```ruby
if (changed & %w{liquid_layout_id liquid_layout}).any?
```
And what about wrapping the statement with `layout_has_changed?`